Python/HAT/hypergraph.py

```
@@ -265,15 +350,20 @@ def set_incidence_matrix(self, IM=None):
-            self._incidence_matrix = np.zeros((self.num_nodes, len(hyperedges)), dtype=int)
+            self._incidence_matrix = np.zeros((self._nodes.shape[0], hyperedges.shape[0]), dtype=int)
             
             # Populate the incidence matrix
             for k, edge in enumerate(hyperedges):
-                for node in edge:
+                for i, node in enumerate(edge):
                     self._incidence_matrix[node, k] = 1  # Node participates in hyperedge `k`
+                    if i == 0:
+                        self._incidence_matrix[node, k] = -1
         else:
             warning.warn("Cannot set the incidence matrix without more information.")
```

Ruff (0.8.0)

Undefined name `warning`

(F821)